src/components/Exchange/ExchangeTVChart.js

```diff
@@ -292,9 +299,9 @@ export default function ExchangeTVChart(props) {
           );
         });
       }
-      if (positions && positions.length > 0) {
+      if (currentPositions && currentPositions.length > 0) {
```

Comment:
mm, looks like currentPositions is always array because you return an empty array in useMemo?